### DISCUSSION TITLE
Using form action redirects

### DISCUSSION THREAD

**Discussion by Tristano8** [react-hook-form/react-hook-form#3265] [Q&A]
I have a form that requires a form redirect to an external page. Using `axios` or `fetch` to return redirects from the server is not quite what I want, because I want to pass along the form inputs as post data like so:

```js
<form action="some-external-endpoint/login" onSubmit={onSubmit)} method="post">
// some inputs
</form>
```

However I would still like to leverage the client-side validation of `react-hook-form`. Is it possible to still do a native form redirect while using `handleSubmit`?

**Comment by bluebill1049** [comment]
you can do your redirect inside the onSubmit function, i did in this section with react router: https://react-hook-form.com/advanced-usage#WizardFormFunnel

**Reply by Tristano8** [reply]
Thanks for the quick reply! I think the client-side router only handles redirects as GET requests, and for security reasons I want the form fields to be sent as POST data. This is possible in a regular HTML form, and the code in my original post actually does what I want (React will do the redirect to the `action` endpoint with post data once `onSubmit` has run). I was just hoping it was also possible to do it programmatically with `handleSubmit` so that I can use `react-hook-form` in-built validation functionality, otherwise I'll have to manually track the state of my inputs to run the validations in the `onSubmit`.

**Reply by bluebill1049** [reply]
what's stopping you from doing `fetch(method: 'POST')`

**Reply by Tristano8** [reply]
It doesn't suit the login flow I was after - the external page I use for the redirect needs to sets some cookies on the user's browser and then forward them to a logged-in page in a separate app. It's possible to set them in the fetch response and then use client-side routing to redirect them to the logged-in page, it just doesn't play as nice with the existing code I need to work with.
